cfme/automate/buttons.py

```diff
@@ -795,19 +802,33 @@ class ButtonGroupCollection(BaseCollection):
     TEMPLATE_IMAGE = "VM Template and Image"
     VM_INSTANCE = "VM and Instance"
 
-    def instantiate(self, text, hover, type, image=None, display=None, icon_color=None):
+    def instantiate(
+        self,
+        text,
+        hover,
+        type,
+        image="fa-user",
```

Comment:
I don't think this is the correct way to define this collection ENTITY's attributes.

1. This entire function definition should be removed, as `BaseCollection.instantiate` covers it.

2. `self.type = type` on L815, we're setting the collection instance `type` attribute on instantiation of a ENTITY. We could instantiate multiple entities on this same collection instance, with different `type` passed each time. I'm not sure at the moment of the purpose of the `ButtonGroupCollection.type` attribute, but this appears to be an inconsistent implementation

3. The default for image that you're trying to define here ('fa-user') contradicts the ENTITY.image `attr.ib()` definition, which does not provide a default. If this entity attribute does not need to be passed during instantiation, then this default should be provided in the ENTITY attibute specification. (L719 above)

Reply:
@mshriver I discuss this  thing before.... nearly  same question. https://github.com/ManageIQ/integration_tests/pull/7964#discussion_r223990458

related to `images`; here already overwrite instantiate method. attr.ib default will not effective then.

If you want I will refactor overall Button model page.... we can even remove object mapping also dependency etc. but, This  implementation working awesome and not even creating problem in automating new test.

Reply:
Do you disagree with 1 or 3 here, or just 2?  Sounds like just 2 needs some discussion, but 1 and 3 are required.

Reply:
For #2, it sounds like we need, in the collection, to store _all_ of the types of buttons that have been instantiated through the collection, not just the _last_ type that was instantiated.

Reply:
Added changes